packages/solana-nft-adapter/test/solanaNft.ts

```diff
@@ -0,0 +1,248 @@
+    // Airdropping tokens to a payer.
+    const latestBlockHash = await provider.connection.getLatestBlockhash();
+
+    // const signature = await provider.connection.requestAirdrop(
+    //   payer.publicKey,
+    //   1.5 * LAMPORTS_PER_SOL
+    // );
+    // await provider.connection.confirmTransaction(
+    //   {
+    //     blockhash: latestBlockHash.blockhash,
+    //     lastValidBlockHeight: latestBlockHash.lastValidBlockHeight,
+    //     signature,
+    //   },
+    //   'confirmed'
+    // );
+    // await sleep(20_000);
+    // console.log('Airdrop signature: ', signature);
+
+    const payerBalance = await provider.connection.getBalance(payer.publicKey);
+    console.log(
+      'Payer ',
+      payer.publicKey.toBase58(),
+      ' balance: ',
+      payerBalance
+    );
+    // Fund Main Accounts
+    const tx = new Transaction();
+    tx.add(
+      SystemProgram.transfer({
+        fromPubkey: payer.publicKey,
+        toPubkey: ownerMainAccount.publicKey,
+        lamports: LAMPORTS_PER_SOL / 10,
+      })
+    );
+    tx.add(
+      SystemProgram.transfer({
+        fromPubkey: payer.publicKey,
+        toPubkey: buyerMainAccount.publicKey,
+        lamports: LAMPORTS_PER_SOL / 10,
+      })
+    );
+    await provider.connection.sendTransaction(tx, [payer]);
+    console.log(
+      'Owner: ',
+      ownerMainAccount.publicKey.toBase58(),
+      ' buyer: ',
+      buyerMainAccount.publicKey.toBase58()
+    );
+    await sleep(20_000);
+    const ownerMainAccountBalance = await provider.connection.getBalance(
+      ownerMainAccount.publicKey
+    );
+    console.log('Owners balance', ownerMainAccountBalance);
+
+    mintKey = await createMint(
+      provider.connection,
+      payer,
+      payer.publicKey,
+      null,
+      0
+    );
+    console.log('mintKey: ', mintKey.toString());
+    ownerTokenAccount = await createAssociatedTokenAccount(
+      provider.connection,
+      payer,
+      mintKey,
+      ownerMainAccount.publicKey
+    );
+    console.log('Owner nft account: ', ownerTokenAccount.toString());
+    await mintTo(
+      provider.connection,
+      payer,
+      mintKey,
+      ownerTokenAccount,
+      payer,
+      1
+    );
+
+    await sleep(10_000);
+    console.log('Here');
+    const _ownerTokenAccount = await getAccount(
+      provider.connection,
+      ownerTokenAccount
+    );
+    console.log(
+      'Owner token account amount: ',
+      Number(_ownerTokenAccount.amount)
+    );
+    assert.ok(Number(_ownerTokenAccount.amount) === 1);
+
+    currencyKey = await createMint(
+      provider.connection,
+      payer,
+      payer.publicKey,
+      null,
+      6
+    );
+    console.log('Here');
+
+    buyerCurrencyAccount = await createAssociatedTokenAccount(
+      provider.connection,
+      payer,
+      currencyKey,
+      buyerMainAccount.publicKey
+    );
+
+    console.log('Buyer currency account: ', buyerCurrencyAccount.toString());
```

Comment:
console logs

Reply:
removed, except one that I need to see balance info when tests running